core/services/run_manager.go

```diff
@@ -257,6 +272,13 @@ func (rm *runManager) Create(
 	ValidateRun(run, runCost)
 
 	if initiator.Type == models.InitiatorRandomnessLog {
+		seenIdentical, err := rm.SeenSameRequestIDAndBlockHashBefore(run)
```

Comment:
There is probably a way to do this in 1 query, but not worth optimizing as we're moving to jpv2 asap